lib/valkyrie/persistence/solr/query_service.rb

```diff
@@ -5,25 +5,26 @@ class QueryService
     attr_reader :connection, :resource_factory
     # @param connection [RSolr::Client]
     # @param resource_factory [Valkyrie::Persistence::Solr::ResourceFactory]
-    def initialize(connection:, resource_factory:)
+    def initialize(connection:, resource_factory:, query_runner: default_query_runner)
       @connection = connection
       @resource_factory = resource_factory
+      @query_runner = query_runner
     end
 
     # (see Valkyrie::Persistence::Memory::QueryService#find_by)
     def find_by(id:)
       validate_id(id)
-      Valkyrie::Persistence::Solr::Queries::FindByIdQuery.new(id, connection: connection, resource_factory: resource_factory).run
+      query_runner.run_find_by_id_query(id, connection: connection, resource_factory: resource_factory)
```

Comment:
The commit before this one headed in a direction that seemed reasonable to me. However, I think now what you've done is create a QueryService which calls a QueryService. Now this one just calls another one with a different method signature.

I wonder if we should move this query service into a path similar to the refactor for the Postgres one: https://github.com/samvera-labs/valkyrie/blob/master/lib/valkyrie/persistence/postgres/query_service.rb

Where there's not an object per query.

Reply:
I defer to your judgement on this; you have a greater context of scope. 

To further clarify, I saw what I consider to be a violation of the Law of Demeter through a chain of calls:

* Find the builder (e.g. `Valkyrie::Persistence::Solr::Queries::FindByIdQuery`)
* Call a builder method (e.g. `.new`)
* Call a query method (e.g. `#call`)

I have had conversations that `Something.new(with: :parameters).call` is a common implementation, but that creates fragility in extension and modification.

I personally don't mind one object per query type, so long as it is below the surface of any public API. In fact, I would consider either tagging those classes with `private_constant` or `@api private` (if they aren't already).